packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -590,3 +602,249 @@ class ThemeData {
   @override
   String toString() => '$runtimeType(${ platform != defaultTargetPlatform ? "$platform " : ''}$brightness $primaryColor etc...)';
 }
+
+/// A lazily evaluated theme that provides the properties of the given
+/// [delegate] theme localized using the properties of the given
+/// [localTextGeometry].
+///
+/// The localization is done by merging of the [TextTheme] fields of the
+/// [delegate] into the [localTextGeometry] and caching the results.
+// Suppressing @immutable warning because this class contains lazily computed
+// fields. Otherwise, for all intents and purposes, this class is immutable.
+// ignore: must_be_immutable
```

Comment:
just remove @immutable on the superclass.

Reply:
Done.